packages/@storefront/flux-capacitor/src/core/actions/validators.ts

```diff
@@ -104,9 +104,15 @@ export const isPastPurchaseRefinementSelectedByIndex: Validator<Actions.Payload.
   msg: 'navigation does not exist or refinement is not selected'
 };
 
-export const isPastPurchasesSortDeselected: Validator<number> = {
-  func: (index, state) => Selectors.pastPurchaseSort(state).selected !== index,
-  msg: 'past purchases sort is already selected'
+export const isPastPurchasesSortValid: Validator<number> = {
+  func: (index, state) => {
+    return (
+      typeof index === 'number'
+      && index >= 0
+      && index <= (Selectors.pastPurchaseSort(state).items.length - 1)
+    );
+  },
+  msg: 'past purchases sort index is not a number or is not within the sorts array'
```

Comment:
Perhaps "... is greater than the length of the sorts array", or a la Java, something something "out of bounds". The idea here is that the index won't be found as a value in the array as the text suggests, it's an index of the array.

Reply:
Done, resolving feedback.